src/attached_probe.cpp

```diff
@@ -960,8 +970,176 @@ void AttachedProbe::attach_kprobe(bool safe_mode)
   perf_event_fds_.push_back(perf_event_fd);
 }
 
+#ifdef HAVE_LIBBPF_UPROBE_MULTI
+struct bcc_sym_cb_data
+{
+  std::vector<std::string> &syms;
+  std::set<uint64_t> &offsets;
+};
+
+static int bcc_sym_cb(const char *symname, uint64_t start, uint64_t, void *p)
+{
+  struct bcc_sym_cb_data *data = static_cast<struct bcc_sym_cb_data *>(p);
+  std::vector<std::string> &syms = data->syms;
+
+  if (std::binary_search(syms.begin(), syms.end(), symname))
+  {
+    data->offsets.insert(start);
+  }
+
+  return 0;
+}
+
+struct addr_offset
+{
+  uint64_t addr;
+  uint64_t offset;
+};
+
+int bcc_load_cb(uint64_t v_addr, uint64_t mem_sz, uint64_t file_offset, void *p)
+{
+  std::vector<struct addr_offset> *addrs =
+      static_cast<std::vector<struct addr_offset> *>(p);
+
+  for (std::vector<struct addr_offset>::iterator it = addrs->begin();
+       it != addrs->end();
+       ++it)
+  {
+    struct addr_offset &a = *it;
```

Comment:
Can we use a range-based loop?
```suggestion
  for (auto &a : addrs)
  {
```

Reply:
ok